test/packages/time_series/data_stream/example/manifest.yml

```diff
@@ -15,3 +15,6 @@ elasticsearch:
     settings:
       # Defaults to 16
       index.mapping.dimension_fields.limit: 32
+      index.routing_path: "example.agent.id"
+    data_stream:
+      index_mode: "time_series"
```

Comment:
> I am not sure about adding this data_stream.index_mode setting to the spec. This setting needs that the data stream has dimension fields, and that index.routing_path is configured to a dimension field of type keyword. If we add this to the spec we should also check that the other requirements are satisfied in order to provide good feedback to package developers. We should also ensure that Fleet reads this setting and installs it in the index template.

I can comment on this one. I'm pretty sure we have a lot of gaps in the validation, but at some point, it helps developers discover what are available properties (`index_mode` with value `time _series`), even though it doesn't validate all dependent places.

Reply:
A few comments/questions:
- Why add a `data_stream.index_mode` field rather than using the existing `index_template.settings.index.mode`?
- What is the dev experience we're aiming for with index settings? Do we want to move towards a more high-level, easy configuration experience or do we prefer making it explicit?
	- If we prefer the latter (which would be my preference), I think we should require the user to set the `index.mode` explicitly and validate this in the spec rather than having Kibana automatically apply the index mode when the setting is present. IMO, the less magic Kibana does to install packages the better.

Reply:
> * Why add a `data_stream.index_mode` field rather than using the existing `index_template.settings.index.mode`?

It seems that after https://github.com/elastic/elasticsearch/pull/82621, it is recommended to use `data_stream.index_mode`. The setting under the index template settings seems to also require to manually set the start and end time of each index.

@martijnvg can probably confirm what setting we should use.

> What is the dev experience we're aiming for with index settings? Do we want to move towards a more high-level, easy configuration experience or do we prefer making it explicit?
>   * If we prefer the latter (which would be my preference), I think we should require the user to set the `index.mode` explicitly and validate this in the spec rather than having Kibana automatically apply the index mode when the setting is present. IMO, the less magic Kibana does to install packages the better.

Not sure what would be the best experience. What concerns me here is that there are three settings that are strongly dependant, a developer needs to use dimensions, set one of these dimensions as routing path and enable time series mode to actually enable time series. I can imagine that a developer can forget of any of these things.

I think that dimensions validation belongs to the spec, there we can statically check if the dimensions and the routing path are valid. But at this point the index mode would be redundant, the developer most likely wants to use it (actually it doesn't seem possible to use a routing path without the time series mode, see https://github.com/elastic/package-spec/pull/297#discussion_r827359257).
I guess that adding another static check for this in the spec is preferred to adding more magic to Kibana.
One problem of this approach is that packages enabling the time series mode will only work with versions of the stack supporting these settings. We make harder for developers to support a broad range of stack versions.

Another, more explicit option, could be to add to the data stream manifest something like this:
```
time_series:
  enabled: true
  routing_path: elastic.agent.id
```
This would need to be explicitly interpreted by Kibana and applied to the index template.

There are two big advantages of this:
* It is more explicit for developers.
* It is more backwards compatible, older versions of the stack could continue using these packages, but without time series features.

Reply:
>  to also require to manually set the start and end time of each index

This is not required, and when setting `index_mode` to `time_series` inside data stream snippet of a template then start and end time index settings are automatically set.

Only the the routing_path index setting needs to be additionally configured as index setting inside the template.

> time_series:
  enabled: true
  routing_path: elastic.agent.id

I like this idea. Assuming that `time_series` is a field inside `data_stream` object of a template?

Reply:
> In any case I think that for out use case we want this to be done automatically, and then it would be in the data stream object of the template, right?

Yes, defining index_mode in data stream object would automatically setup the start and end time index settings for the indices that are backing a data stream.

> Here I am talking about the data_stream manifest in a package (like [this one for apache status](https://github.com/elastic/integrations/blob/a987263b681b4c28322770a49d80fadeb8cacb9b/packages/apache/data_stream/status/manifest.yml)). This would need to be translated by Fleet to the proper template settings when installing a package.

I see. I'm currently discussing with others to change how tsdb data streams are configured in templates. In the way that you describe. (bringing routing_path index setting into the data stream object of a template)